src/pages/layout/Search/Index.tsx

```diff
@@ -63,157 +64,195 @@ export default function HeaderSearch() {
     // eslint-disable-next-line react-hooks/exhaustive-deps
   }, [inputValue]);
 
+  async function handleAnsName(
+    searchText: string,
+  ): Promise<SearchResult | null> {
+    try {
+      const name = await state.sdk_v2_client.getName({
+        name: searchText,
+      });
+
+      const address = name?.registered_address ?? name?.owner_address;
+
+      if (name && address) {
+        return {
+          label: `Account ${truncateAddress(address)} ${searchText}`,
+          to: `/account/${address}`,
+        };
+      }
+    } catch (e) {}
+
+    return null;
+  }
+
+  async function handleCoin(searchText: string): Promise<SearchResult | null> {
+    const address = searchText.split("::")[0];
+    return getAccountResource(
+      {address, resourceType: `0x1::coin::CoinInfo<${searchText}>`},
+      state.aptos_client,
+    ).then(
+      () => {
+        return {
+          label: `Coin ${searchText}`,
+          to: `/coin/${searchText}`,
+        };
+      },
+      () => {
+        return null;
+      },
+    );
+  }
```

Comment:
nit: You don't need to do it, but would be nice to stay consistent to either the `promise.then.catch` route or the `try {} catch {}` route you have above.